hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -82,17 +90,38 @@
 import static java.lang.String.format;
 
 /**
- * Validates a Hazelcast configuration in a specific context like OS vs. EE or client vs. member nodes.
+ * Validates a Hazelcast configuration in a specific
+ * context like OS vs. EE or client vs. member nodes.
  */
+@SuppressWarnings("checkstyle:classfanoutcomplexity")
 public final class ConfigValidator {
 
-    private static final ILogger LOGGER = Logger.getLogger(ConfigValidator.class);
+    /**
+     * By default, hot restart requires at least 15% free HD space.
+     */
+    public static final int DEFAULT_HOT_RESTART_FREE_NATIVE_MEMORY_PERCENTAGE = 15;
+    /**
+     * The name of the XML and system property for setting the hot restart required free space.
+     */
+    public static final String PROPERTY_HOTRESTART_FREE_NATIVE_MEMORY_PERCENTAGE =
+            "hazelcast.hotrestart.free.native.memory.percentage";
+    /**
+     * XML and system property for setting the hot restart required free space
+     */
+    public static final HazelcastProperty HOT_RESTART_FREE_NATIVE_MEMORY_PERCENTAGE = new HazelcastProperty(
```

Comment:
Feels strange seeing group properties in `ConfigValidator`. Maybe move to `GroupProperty`?

Reply:
It was effectively like this in EE code, just kept it. But i also think group-property file could be a better place, now i moved it there.